Content.Client/Silicons/Borgs/BorgSwitchableTypeSystem.cs

```diff
@@ -0,0 +1,88 @@
+﻿using Content.Shared.Movement.Components;
+using Content.Shared.Silicons.Borgs;
+using Content.Shared.Silicons.Borgs.Components;
+using Robust.Client.GameObjects;
+using Robust.Client.ResourceManagement; // DeltaV
+using static Robust.Shared.Serialization.TypeSerializers.Implementations.SpriteSpecifierSerializer; // DeltaV
+
+namespace Content.Client.Silicons.Borgs;
+
+/// <summary>
+/// Client side logic for borg type switching. Sets up primarily client-side visual information.
+/// </summary>
+/// <seealso cref="SharedBorgSwitchableTypeSystem"/>
+/// <seealso cref="BorgSwitchableTypeComponent"/>
+public sealed class BorgSwitchableTypeSystem : SharedBorgSwitchableTypeSystem
+{
+    [Dependency] private readonly BorgSystem _borgSystem = default!;
+    [Dependency] private readonly AppearanceSystem _appearance = default!;
+    [Dependency] private readonly IResourceCache _resourceCache = default!; // DeltaV
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<BorgSwitchableTypeComponent, AfterAutoHandleStateEvent>(AfterStateHandler);
+        SubscribeLocalEvent<BorgSwitchableTypeComponent, ComponentStartup>(OnComponentStartup);
+    }
+
+    private void OnComponentStartup(Entity<BorgSwitchableTypeComponent> ent, ref ComponentStartup args)
+    {
+        UpdateEntityAppearance(ent);
+    }
+
+    private void AfterStateHandler(Entity<BorgSwitchableTypeComponent> ent, ref AfterAutoHandleStateEvent args)
+    {
+        UpdateEntityAppearance(ent);
+    }
+
+    protected override void UpdateEntityAppearance(
+        Entity<BorgSwitchableTypeComponent> entity,
+        BorgTypePrototype prototype)
+    {
+        if (TryComp(entity, out SpriteComponent? sprite))
+        {
+            // DeltaV: Begin sprite path stuff
+            if (prototype.SpritePath is {} path && _resourceCache.TryGetResource(TextureRoot / path, out RSIResource? resource))
+                sprite.BaseRSI = resource.RSI;
+            // DeltaV: End sprite path stuff
+            sprite.LayerSetState(BorgVisualLayers.Body, prototype.SpriteBodyState);
+            sprite.LayerSetState(BorgVisualLayers.LightStatus, prototype.SpriteToggleLightState);
```

Comment:
@MilonPL need help with this

if it sets the rsi first then the layers are wrong
if it sets the layers first they point to invalid states

should i just make secborg be security.rsi and use robot* states so it Just Works?

Reply:
hold on lemme look into it

Reply:
I'M SCREAMING INTO THE VOID AND IT'S NOT LISTENING
i ran out of fucking ideas, it doesn't want to work,
this is *an* attempt but it's a giant fucking meme
sort of works but I think the Light layer is wrong? not sure if it works correctly but you can test it yourself
```cs
    protected override void UpdateEntityAppearance(
        Entity<BorgSwitchableTypeComponent> entity,
        BorgTypePrototype prototype)
    {
        if (!TryComp(entity, out SpriteComponent? sprite))
            return;

        sprite.LayerMapRemove(BorgVisualLayers.Body);
        sprite.LayerMapRemove(BorgVisualLayers.LightStatus);
        sprite.LayerMapRemove(BorgVisualLayers.Light);

        for (var i = sprite.AllLayers.Count() - 1; i >= 0; i--)
        {
            sprite.RemoveLayer(i);
        }

        if (prototype.SpritePath is {} path)
        {
            if (_resourceCache.TryGetResource(TextureRoot / path, out RSIResource? resource))
            {
                sprite.BaseRSI = resource.RSI;
            }
        }

        var bodyIndex = sprite.AddBlankLayer();
        var lightStatusIndex = sprite.AddBlankLayer();
        var lightIndex = sprite.AddBlankLayer();

        sprite.LayerMapSet(BorgVisualLayers.Body, bodyIndex);
        sprite.LayerMapSet(BorgVisualLayers.LightStatus, lightStatusIndex);
        sprite.LayerMapSet(BorgVisualLayers.Light, lightIndex);

        sprite.LayerSetState(BorgVisualLayers.Body, prototype.SpriteBodyState);
        sprite.LayerSetState(BorgVisualLayers.LightStatus, prototype.SpriteToggleLightState);

        sprite.LayerSetState(BorgVisualLayers.Light, prototype.SpriteNoMindState);
        sprite.LayerSetVisible(BorgVisualLayers.Light, false);

        if (prototype.SpriteBodyMovementState is { } movementState)
        {
            var spriteMovement = EnsureComp<SpriteMovementComponent>(entity);
            spriteMovement.NoMovementLayers.Clear();
            spriteMovement.NoMovementLayers["movement"] = new PrototypeLayerData
            {
                State = prototype.SpriteBodyState,
            };
            spriteMovement.MovementLayers.Clear();
            spriteMovement.MovementLayers["movement"] = new PrototypeLayerData
            {
                State = movementState,
            };
        }
        else
        {
            RemComp<SpriteMovementComponent>(entity);
        }

        // Update chassis mind states if present
        if (TryComp(entity, out BorgChassisComponent? chassis))
        {
            _borgSystem.SetMindStates(
                (entity.Owner, chassis),
                prototype.SpriteHasMindState,
                prototype.SpriteNoMindState);

            if (TryComp(entity, out AppearanceComponent? appearance))
            {
                _appearance.QueueUpdate(entity, appearance);
            }
        }

        base.UpdateEntityAppearance(entity, prototype);
    }
```

Reply:
also it throws fucking errors anyways related to the movement for janiborg and othersTM anyways idfk